protocol/synthetix/contracts/storage/AccountDelegationIntents.sol

```diff
@@ -0,0 +1,149 @@
+
+    /**
+     * @dev Returns the account delegation intents stored at the specified account id.
+     */
+    function loadValid(uint128 id) internal view returns (Data storage accountDelegationIntents) {
+        accountDelegationIntents = load(id);
+        if (accountDelegationIntents.accountId != 0 && accountDelegationIntents.accountId != id) {
+            revert InvalidAccountDelegationIntents();
+        }
+    }
+
+    /**
+     * @dev Returns the account delegation intents stored at the specified account id. Checks if it's valid
+     */
+    function getValid(uint128 id) internal returns (Data storage accountDelegationIntents) {
+        accountDelegationIntents = loadValid(id);
+        if (accountDelegationIntents.accountId == 0) {
+            // Uninitialized storage will have a 0 accountId
+            accountDelegationIntents.accountId = id;
+        }
+    }
+
+    function addIntent(Data storage self, DelegationIntent.Data storage delegationIntent) internal {
+        self.intentsId.add(delegationIntent.id);
+        self
+            .intentsByPair[
+                keccak256(abi.encodePacked(delegationIntent.poolId, delegationIntent.accountId))
+            ]
+            .add(delegationIntent.id);
+
+        self.netDelegatedAmountPerCollateral[delegationIntent.collateralType] += delegationIntent
+            .deltaCollateralAmountD18;
+
+        if (!self.delegatedCollaterals.contains(delegationIntent.collateralType)) {
+            self.delegatedCollaterals.add(delegationIntent.collateralType);
+        }
+    }
+
+    function removeIntent(
+        Data storage self,
+        DelegationIntent.Data storage delegationIntent
+    ) internal {
+        if (!self.intentsId.contains(delegationIntent.id)) {
+            return;
+        }
+
+        self.intentsId.remove(delegationIntent.id);
+        self
+            .intentsByPair[
+                keccak256(abi.encodePacked(delegationIntent.poolId, delegationIntent.accountId))
+            ]
+            .remove(delegationIntent.id);
+
+        self.netDelegatedAmountPerCollateral[delegationIntent.collateralType] -= delegationIntent
+            .deltaCollateralAmountD18;
+    }
+
+    function getIntent(
+        Data storage self,
+        uint256 intentId
+    ) internal view returns (DelegationIntent.Data storage) {
+        if (!self.intentsId.contains(intentId)) {
+            revert DelegationIntent.InvalidDelegationIntentId();
+        }
+        return DelegationIntent.load(intentId);
+    }
+
+    /**
+     * @dev Returns the delegation intent stored at the specified nonce id.
+     */
+    function intentIdsByPair(
+        Data storage self,
+        uint128 poolId,
+        uint128 accountId
+    ) internal view returns (uint256[] memory intentIds) {
+        return self.intentsByPair[keccak256(abi.encodePacked(poolId, accountId))].values();
+    }
+
+    /**
+     * @dev Cleans all intents related to the account. This should be called upon liquidation.
```

Comment:
Probably this isn't called upon liquidation, can be called when you want to clean up an account?

Reply:
fixed the comment